externals/kyuubi-spark-sql-engine/pom.xml

```diff
@@ -255,6 +255,15 @@
                             <include>org.checkerframework:checker-qual</include>
                             <include>org.codehaus.mojo:animal-sniffer-annotations</include>
                         </includes>
+                        <excludes>
+                            <exclude>io.netty:netty-codec-*-quic</exclude>
+                            <exclude>io.netty:netty-resolver-dns-*-macos</exclude>
+                            <exclude>io.netty:netty-tcnative-boringssl-static:*:osx-*</exclude>
+                            <exclude>io.netty:netty-tcnative-boringssl-static:*:windows-*</exclude>
+                            <exclude>io.netty:netty-transport-*-kqueue</exclude>
+                            <exclude>io.netty:netty-transport-*-io_uring</exclude>
+                            <exclude>io.netty:netty-transport-native-epoll:*:linux-riscv64</exclude>
+                        </excludes>
```

Comment:
weirdly, maven resolves those transitive deps of `spark-sql`(provided scope) as runtime scope, so I must exclude them ...

Reply:
@turboFei we can't do that.

currently, kyuubi pulls a few netty deps from etcd-java, which uses gRPC and thus depends on grpc-netty. spark uses more netty deps for intenral rpc